core/chains/evm/gas/rollups/op_l1_oracle.go

```diff
@@ -88,6 +90,19 @@ const (
 )
 
 func NewOpStackL1GasOracle(lggr logger.Logger, ethClient l1OracleClient, chainType chaintype.ChainType, daOracle evmconfig.DAOracle) (*optimismL1Oracle, error) {
+	if daOracle.OracleType() == nil {
+		return nil, errors.New("OracleType is required but was nil")
+	}
+	if *daOracle.OracleType() != toml.DAOracleOPStack {
+		return nil, fmt.Errorf("expected %s oracle type, got %s", toml.DAOracleOPStack, *daOracle.OracleType())
+	}
+	if daOracle.OracleAddress() == nil || *daOracle.OracleAddress() == "" {
+		return nil, errors.New("OracleAddress is required but was nil or empty")
+	}
+	if daOracle.CustomGasPriceCalldata() != nil && *daOracle.CustomGasPriceCalldata() != "" {
+		lggr.Warnf("CustomGasPriceCalldata is set but will be ignored for OPStack DA oracle")
+	}
```

Comment:
It is fine to have validation here, but this typically does not give very good user feedback. Does the toml config validation include this logic? We can report better contextual information from there.

Reply:
Added this validation to `func (o *DAOracle) ValidateConfig() (err error)` in `core/chains/evm/config/toml/config.go`